src/robotFactory.js

```diff
@@ -1,17 +1,72 @@
 'use strict';
 
-function BaseRobot() {
-  // implement
+function BaseRobot(name, weight, coords, chipVersion) {
+  this.name = name;
+  this.weight = weight;
+  this.coords = coords;
+  this.chipVersion = chipVersion;
 }
 
-function FlyingRobot() {
-  // implement
+BaseRobot.prototype.goForward = function(step = 1) {
+  this.coords.y += step;
+};
+
+BaseRobot.prototype.goBack = function(step = 1) {
+  this.coords.y -= step;
+};
+
+BaseRobot.prototype.goLeft = function(step = 1) {
+  this.coords.x -= step;
+};
+
+BaseRobot.prototype.goRight = function(step = 1) {
+  this.coords.x += step;
+};
+
+BaseRobot.prototype.getInfo = function() {
+  return `Robot: ${this.name}, Chip version:\
+ ${this.chipVersion}, Weight: ${this.weight}`;
+};
+
+function FlyingRobot(name, weight, coords, chipVersion) {
+  BaseRobot.call(this, name, weight, coords, chipVersion);
+  this.coords.z = 0;
 }
 
-function DeliveryDrone() {
-  // implement
+FlyingRobot.prototype.goUp = function(step = 1) {
+  this.coords.z += step;
+};
+
+FlyingRobot.prototype.goDown = function(step = 1) {
+  this.coords.z -= step;
+};
+
+Object.setPrototypeOf(FlyingRobot.prototype, BaseRobot.prototype);
+
+function DeliveryDrone(
+  name,
+  weight,
+  coords,
+  chipVersion,
+  maxLoadWeight,
+  currentLoad
+) {
+  this.maxLoadWeight = maxLoadWeight;
+  this.currentLoad = currentLoad;
 }
 
+DeliveryDrone.prototype.hookLoad = function(cargo) {
+  if (cargo.weight <= this.maxLoadWeight && this.currentLoad === null) {
+    this.currentLoad = cargo;
+  }
+};
+
+DeliveryDrone.prototype.unhookLoad = function() {
+  this.currentLoad = null;
+};
```

Comment:
можно написать все методы в одном прототипе
```suggestion
DeliveryDrone.prototype = {
  hookLoad(cargo) {
    if (cargo.weight <= this.maxLoadWeight && this.currentLoad=== null) {
      this.currentLoad = cargo;
    }
  },

  unhookLoad() {
    this.currentLoad = null;
  },
};
```

Reply:
thanks! done